lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -988,6 +1057,31 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
         Ok(())
     }
 
+    fn add_token_token_program(&mut self, c: Context<ConstraintTokenProgram>) -> ParseResult<()> {
+        if self.token_token_program.is_some() {
+            return Err(ParseError::new(
+                c.span(),
+                "token token_program already provided",
+            ));
+        }
+        self.token_token_program.replace(c);
+        Ok(())
+    }
+
+    fn add_associated_token_program(
```

Comment:
Same here, consistency with `add_token_token_program` above and to make it more clear.

```suggestion
    fn add_associated_token_token_program(
```

Reply:
Yep, good spot